yet_another_retnet/retention.py

```diff
@@ -216,9 +216,9 @@ def retention_chunkwise(
         torch.arange(key.size(2), device=key.device, dtype=key.dtype) + 1,
         "n -> () () n ()",
     )
-    states = einsum(key, value, "b h n d1, b h n d2 -> b h n d1 d2")
     state_decays = decay_gammas ** (key.size(2) - inner_pos)
-    state = einsum(states, state_decays, "b h n d1 d2, _ h n _ -> b h d1 d2")
+    discounted_key = einsum(key, state_decays, 'b h n d, _ h n _ -> b h n d')
```

Comment:
This is awesome 🔥 

Great find, and thanks for another excellent PR!

Reply:
Thank you @fkodom, your repo is the best RetNet implementation out there in my opinion! 🔥